include/fsm/fill_spill_merge.hpp

```diff
@@ -511,14 +515,17 @@ static dh_label_t OverflowInto(
   //it's time to stop. (This may be the leaf node of another metadepression, the
   //ocean, or a standard node.)
   if(root==stop_node){                   //We've made a loop, so everything is full
-    if(this_dep.parent==OCEAN)           //If our parent is the ocean
+    if(this_dep.parent==OCEAN){           //If our parent is the ocean
+      this_dep.water_vol = std::min(this_dep.dep_vol,this_dep.water_vol + extra_water);
```

Comment:
Why?

Reply:
this_dep is the depression that should be receiving the current round of extra_water. Here, we are checking to see if the following parent up is the ocean, because that should mean that all additional water should overflow into the ocean and disappear. However, we have never actually added the water to this_dep! This_dep may be completely empty or only partially full and still needs to receive this water. If you look in the 'else' that immediately follows, you will see that this_dep receives the water there, but not here. 
Actually, as I am typing this response, I realise that even if this_dep.parent == OCEAN, we are not giving this_dep a chance to overflow to its sibling depression in this case, so perhaps still some work to be done here.

Reply:
I think this change is solid. I'm pretty sure the `root==stop_node` means that there has been a chance to overflow into the sibling and we're coming out of that traversal at this point.

Reply:
There was a chance to pass to the original depression that first called the function's sibling; we have now reached that original depression's parent. But what about the parent's sibling? I now think we should simply have 
if(root==stop_node){
    this_dep.water_vol += extra_water;
    return stop_node;
}

If the current this_dep (i.e. the original depression's parent) then calls the function again, and its sibling is full, and its parent is the ocean, this will be handled in the 'if' above where we do an ocean check.

Reply:
I think this last iteration is correct. Ideally, you would update the PR to reflect the discussion, but I'm pulling in manually.